# Project 2 -- Niharika Rai

**TA Help:** (for instance) John Smith, Alice Jones, etc., list names of any TAs who helped you

- For example: Help with figuring out how to write a function (describe the tasks that they helped you with)

**Collaboration:** My Friend in CS, My Uncle, Another Student, etc., list names of any other people who helped you

(describe the tasks that they helped you with)
- For example: helped figuring out how to load the dataset.
- Another example: helped debug error with my plot.

**Internet Resources:** Stack Exchange, Stack Overflow, etc.

(describe any information that you learned from internet resources, including the URLs)
- data frames in Pandas versus R from StackOverflow  https://stackoverflow.com/questions/8991709/why-were-pandas-merges-in-python-faster-than-data-table-merges-in-r-in-2012

**ChatGPT, Gemini, Claude, etc:** Any language models or generative AI chatbots that helped you.

(if you used any such tools, please tell us here)
- For example:  I asked ChatGPT how to define a new data frames
- Another example:  Gemini told me how to make a function for sorting my data

**OVERALL MESSAGE:** Any time that you used anything except your brain to solve the questions in these projects, you need to disclose such resources at the start of the project, with details about your usage of the tools.

**YOUR OWN WORK:** Even when you utilize other resources, do NOT just copy and paste.  Write all explanations in your own words, using several sentences in English, which are understandable and which you wrote (and did not just copy and paste).

## Question 1

In [29]:
%%bash 
echo $HOME
pwd

/home/x-nrai1
/home/x-nrai1


In [34]:
%%bash 
cd /anvil/projects/tdm/data
pwd

/anvil/projects/tdm/data


In [31]:
%%bash 
echo $HOME
pwd

/home/x-nrai1
/home/x-nrai1


The observations in steps 1 and 2 are different. Step 1 printed the home directory and step 2 printed the anvil/projects/tdm/data directory. Based on these outputs, I feel that HOME displays the personal home directory, and pwd displays the directory that you are currently working with. Hence, $HOME will always print the home directory, but pwd can print the directory that is different from home if you are working on another directory. 

## Question 2

In [55]:
%%bash 
cd /anvil/projects/tdm/data/zillow       
pwd

/anvil/projects/tdm/data/zillow


In [54]:
%%bash 
cd /anvil/projects/tdm/data/zillow      
cd  
pwd

/home/x-nrai1


In [39]:
%%bash 
cd /anvil/projects/tdm/data/zillow
cd . 
pwd

/anvil/projects/tdm/data/zillow


In [40]:
%%bash 
cd /anvil/projects/tdm/data/zillow
cd ..
pwd

/anvil/projects/tdm/data


In [41]:
%%bash 
cd /anvil/projects/tdm/data/zillow
cd ../../
pwd

/anvil/projects/tdm


In [93]:
%%bash 
cd /anvil/projects/tdm/data/zillow
ls 
pwd

City_time_series.csv
CountyCrossWalk_Zillow.csv
County_time_series.csv
DataDictionary.csv
Metro_time_series.csv
Neighborhood_time_series.csv
State_time_series.csv
Zip_time_series.csv
all_available_metrics.json
cities_crosswalk.csv
fields_per_level.json
/anvil/projects/tdm/data/zillow


In [94]:
%%bash 
cd /anvil/projects/tdm/data/zillow
ls .
pwd

City_time_series.csv
CountyCrossWalk_Zillow.csv
County_time_series.csv
DataDictionary.csv
Metro_time_series.csv
Neighborhood_time_series.csv
State_time_series.csv
Zip_time_series.csv
all_available_metrics.json
cities_crosswalk.csv
fields_per_level.json
/anvil/projects/tdm/data/zillow


In [48]:
%%bash 
cd /anvil/projects/tdm/data/zillow
ls -la 
pwd

total 1871363
drwxrwsr-x+  2 nobody nogroup      4096 Aug  5  2023 .
drwxrwsr-x+ 74 nobody nogroup      4096 Aug 11 16:42 ..
-rw-r--r--   1 nobody nogroup 689627012 Sep 21  2019 City_time_series.csv
-rw-r--r--   1 nobody nogroup    232467 Sep 21  2019 CountyCrossWalk_Zillow.csv
-rw-r--r--   1 nobody nogroup 112778579 Sep 21  2019 County_time_series.csv
-rw-r--r--   1 nobody nogroup      4668 Sep 21  2019 DataDictionary.csv
-rw-r--r--   1 nobody nogroup  56232114 Sep 21  2019 Metro_time_series.csv
-rw-r--r--   1 nobody nogroup 265790699 Sep 21  2019 Neighborhood_time_series.csv
-rw-r--r--   1 nobody nogroup   4742679 Sep 21  2019 State_time_series.csv
-rw-r--r--   1 nobody nogroup 782137407 Sep 21  2019 Zip_time_series.csv
-rw-r--r--   1 nobody nogroup      3409 Sep 21  2019 all_available_metrics.json
-rw-r--r--   1 nobody nogroup   1074309 Sep 21  2019 cities_crosswalk.csv
-rw-r--r--   1 nobody nogroup     17005 Sep 21  2019 fields_per_level.json
/anvil/projects/tdm/data/zillow


In [49]:
%%bash 
cd /anvil/projects/tdm/data/zillow
ls -la .  
pwd

total 1871363
drwxrwsr-x+  2 nobody nogroup      4096 Aug  5  2023 .
drwxrwsr-x+ 74 nobody nogroup      4096 Aug 11 16:42 ..
-rw-r--r--   1 nobody nogroup 689627012 Sep 21  2019 City_time_series.csv
-rw-r--r--   1 nobody nogroup    232467 Sep 21  2019 CountyCrossWalk_Zillow.csv
-rw-r--r--   1 nobody nogroup 112778579 Sep 21  2019 County_time_series.csv
-rw-r--r--   1 nobody nogroup      4668 Sep 21  2019 DataDictionary.csv
-rw-r--r--   1 nobody nogroup  56232114 Sep 21  2019 Metro_time_series.csv
-rw-r--r--   1 nobody nogroup 265790699 Sep 21  2019 Neighborhood_time_series.csv
-rw-r--r--   1 nobody nogroup   4742679 Sep 21  2019 State_time_series.csv
-rw-r--r--   1 nobody nogroup 782137407 Sep 21  2019 Zip_time_series.csv
-rw-r--r--   1 nobody nogroup      3409 Sep 21  2019 all_available_metrics.json
-rw-r--r--   1 nobody nogroup   1074309 Sep 21  2019 cities_crosswalk.csv
-rw-r--r--   1 nobody nogroup     17005 Sep 21  2019 fields_per_level.json
/anvil/projects/tdm/data/zillow


In [50]:
%%bash 
cd /anvil/projects/tdm/data/zillow
ls ../
pwd

8451
CMS
DELETE_house-prices
DELETE_wine
aiannh
airbnb
amazon
apple
archive
bay_area_bike_share
beer
bosch
boston_housing
boundaries
chinook
co2
coco
concrete_crack_data_tfrecords
concrete_cracks_mini
consumer_complaints
craigslist
death_records
disney
donorschoose
election
expedia
fao
fars
fips
flights
forest
goodreads
grouping
hidden
icecream
images
iowa_liquor_sales
iris
lahman
mortgage_rates
movies_and_tv
noaa
noaa_timeseries
noaa_timeseries2
nps
nyc
okcupid
olympics
open_food_facts
otc
precip
project05
project09
ravdess
restaurant
scavenger_hunt
shopper_intention
sim
social_media_addiction
spotify
stackoverflow
starwars
taxi
techcrunch
tippecanoe
whin
wine_quality
worldbank
yelp
youtube
zillow
/anvil/projects/tdm/data/zillow


In [58]:
%%bash 
cd /anvil/projects/tdm/data/zillow
pwd

/anvil/projects/tdm/data/zillow


2d. Written Explanation of each command. 

cd - Brings us back to the home directory. 
cd . - Tells us to stay in the same directory, which is the /zillow directory. 
cd .. - Takes us 1 directory up to the parent directory. In this case, it takes us to the /data directory. 
cd ../../ - Takes us 2 directories up to the parent directory. In this case, it takes us to the /tdm directory.
ls or ls . - Both give the same output, that is they display the file names in the /zillow directory. 
ls -la or ls -la . - Both give the same output, that is they display more information about the file contents in the /zillow directory. 
ls ../ - This displays the names of the subdirectories in the parent directory that is 1 level up from the current directory. Hence, this is the /data directory content names. However, the pwd command shows that our current directory is still the /zillow directory. 

2e. From my observation of the code outputs, I believe that using relative paths for particular commands like ls can benefit us by allowing more flexibility in viewing file names and contents of parent directories, while staying the same directory. For example, using just ls would give the file names of the present directory. But, using ls ../ allows us to explore the contents of the parent directory without changing your current directory.  

## Question 3

In [1]:
%%bash 
cd /anvil/projects/tdm/data/icecream/breyers
pwd

/anvil/projects/tdm/data/icecream/breyers


In [2]:
%%bash 
head -n5 /anvil/projects/tdm/data/icecream/breyers/reviews.csv

key,author,date,stars,title,helpful_yes,helpful_no,text
0_breyers,LABEL READER,2020-04-17,2,EVER INCREASING INGEDIENTS LIST.,81,3,"Not the long ago Breyers set itself apart from other brands by using ONLY MILK, CREAM, SUGAR, and VANILLA in it's Vanilla ice cream. They went so far as to have TV commercials with children reading the ver very simple label. Now they show similar comercials stating the Breyers is made WITH those ingredients but no longer ONLY WITH. Read the label for yourself. The ice cream is still good but the longer ingrediants list ant the missing word ""ONLY"" hasn't gone unnoticed despite the attempt to gloss over the corporate decision to move away from the one distinctive characteristic that set the Breyers brand apart from the others."
0_breyers,AK in WA,2020-06-09,1,This is NO LONGER the Breyers We Used to Know,94,2,"I must agree with many one-star ratings here. The main beef is the huge change in your formula/ingredients. It's always been Breyers. When I first mo

In [3]:
%%bash 
tail -n5 /anvil/projects/tdm/data/icecream/breyers/reviews.csv

68_breyers,Randi,2020-04-22,1,Terrible missing the chocolate ice cream,0,0,"There was no chocolate ice cream in this at all, this was very very disappointing, didn’t even come close to looking like the package"
68_breyers,Bethie,2020-09-08,1,Terrible,0,0,This ice cream has no flavor at all. No one in the entire family liked it. Kids refusing ice cream - for real?
68_breyers,Nshaw1994,2020-09-09,5,Flavor of the week!,0,0,Absolutely love this flavor! The only thing that would make it better is caramel and cookie dough bites!
68_breyers,PamelaG,2020-06-01,5,Love this flavor!!,0,0,Brilliant combo - love the cheesecake and brownie chunks - please keep making this!!
68_breyers,Fanchon,2020-08-22,5,Great,0,0,Has a delicious taste with all natural ingredients


In [4]:
%%bash 
head -n1 /anvil/projects/tdm/data/icecream/breyers/reviews.csv

key,author,date,stars,title,helpful_yes,helpful_no,text


In [7]:
%%bash
wc /anvil/projects/tdm/data/icecream/breyers/reviews.csv

   5464  225282 1436490 /anvil/projects/tdm/data/icecream/breyers/reviews.csv


In [13]:
%%bash
wc -l /anvil/projects/tdm/data/icecream/breyers/reviews.csv

5464 /anvil/projects/tdm/data/icecream/breyers/reviews.csv


I followed Dr. Ward's video instructions for this question to print the first 5, last 5 rows, column names, and the line count in the dataset. 

3c. The different parts of the wc command output are given below. I used ChatGPT to find out what 225282 and 1436490 represented in this code output. 

5464 - This is the line count. 
225282 - This is the word count. 
1436490 - This is the number of bytes/characters in the dataset. 
/anvil/projects/tdm/data/icecream/breyers/reviews.csv - This is the filename. 

## Question 4

In [51]:
%%bash
cp /anvil/projects/tdm/data/icecream/breyers/reviews.csv $SCRATCH

In [52]:
%%bash
cp -r /anvil/projects/tdm/data/icecream $SCRATCH

In [53]:
%%bash 
ls $SCRATCH

icecream
reviews.csv


In [54]:
%%bash 
rm $SCRATCH/reviews.csv 

In [55]:
%%bash 
rm -r $SCRATCH/icecream

In [56]:
%%bash 
ls $SCRATCH

I watched Dr. Ward's videos for reference on how to use the cp, rm, and rm -r commands. The -r is the recursive flag, which is needed to remove or copy the entire directory, such as icecream. I learnt this fact from ChatGPT. I used the ls command to view the file names in the scratch directory. After I deleted the reviews.csv and icecream directory, I used the ls command again to check, and nothing was printed. This indicated that the CSV file and directory had been deleted, and those 2 were the only things in my scratch directory (it is now empty). 

4a. The size of the home directory is 11.7MB, and its limit is 25.0GB (I typed 'myquota' in the terminal to find these numbers) 
The size of the scratch directory is 0.0KB, and its limit is 100.0TB (The code outputs were not found using Jupyter Book. I used Terminal instead, based on the question instructions.)  

## Question 5

In [73]:
%%bash 
mkdir $HOME/mydinner

In [76]:
%%bash 
cd $HOME/mydinner 
touch spaghetti.txt bread.txt broccoli.txt smoothie.txt tiramisu.txt

In [77]:
%%bash 
ls $HOME/mydinner 

bread.txt
broccoli.txt
smoothie.txt
spaghetti.txt
tiramisu.txt


In [84]:
%%bash 
cd $HOME/mydinner 

echo "noodle, tomato sauce" > spaghetti.txt
echo "bread, garlic, butter, cheese" > bread.txt
echo "broccoli, salt, pepper" > broccoli.txt
echo "strawberry, banana, milk" > smoothie.txt
echo "top-secret tiramisu recipe from granny" > tiramisu.txt

In [85]:
%%bash 
cd $HOME/mydinner  
cat *.txt

bread, garlic, butter, cheese
broccoli, salt, pepper
strawberry, banana, milk
noodle, tomato sauce
top-secret tiramisu recipe from granny


In [86]:
%%bash 
mv $HOME/mydinner $SCRATCH/mybreakfast

In [89]:
%%bash 
ls $SCRATCH

mybreakfast


In [91]:
%%bash 
rm -r $SCRATCH/mybreakfast

In [92]:
%%bash 
ls $SCRATCH

I used Dr. Ward's previous videos for question 4 (mkdir and rm -r commands) and ChatGPT to learn how to use the touch, cat, and mv commands, and the > symbol. My code outputs are shown above. I already knew about echo, cd, and ls from the previous questions and project 1. 

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.

https://www.purdue.edu/odos/osrr/honor-pledge/
